cmd/podman/ps.go

```diff
@@ -316,8 +325,21 @@ func generateContainerFilterFuncs(filter, filterValue string, runtime *libpod.Ru
 		//- volume=(<volume-name>|<mount-point-destination>)
 		return func(c *libpod.Container) bool {
 			containerConfig := c.Config()
-			//TODO We need to still lookup against volumes too
-			return containerConfig.MountLabel == filterValue
+			var dest string
+			arr := strings.Split(filterValue, ":")
+			source := arr[0]
+			if len(arr) == 2 {
+				dest = arr[1]
+			}
+			for _, mount := range containerConfig.Spec.Mounts {
+				if dest != "" && (mount.Source == source && mount.Destination == dest) {
+					return true
+				}
+				if mount.Source == source {
```

Comment:
Should we match this only if dest was not specified?

Reply:
Right, that is what the first if (line 335) is doing. If dest is specified it checks for a match for both source and destination. Otherwise just checks for a match for source.

Reply:
But if that doesn't match, we check again based on only the source, even if a dest was given. Shouldn't we have a && dest == "" here?

Reply:
Woops you are right! Will add in that check.

Reply:
fixed.